testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,25 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    sub = u.atoms[:100]
+    #ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
```

Comment:
This should be one test function.

Each test should only create one `ans`.

Reply:
You might be able to write the three tests as a parametrized test with `@pytest.mark.paramterize` – however, if you haven't done this before, just write three separate tests to see if there's enough common structure and then we can help with condensing in a parametrized test.